worldengine/simulations/permeability.py

```diff
@@ -19,20 +20,18 @@ def execute(self, world, seed):
 
     @staticmethod
     def _calculate(seed, width, height):
-        random.seed(seed * 37)
-        base = random.randint(0, 4096)
-        perm = [[0 for x in range(width)] for y in range(height)]  # TODO: replace with numpy
+        rng = numpy.random.RandomState(seed)  # create our own random generator
```

Comment:
Instead of permanently modifying the global RNG (e.g. numpy.random.randint()) the sub-steps now use their own RNG whenever easily possible (and they don't modify the seeds anymore due to that now happening in generation.py).

This makes the main generation mostly independent of the order in which steps are done. In addition, should a GUI be ever able to let the user modify seeds for different steps of the generation, that seed would then actually be used instead of being multiplied by an arbitrary number.